RA-16

KinCadeFoster opened the pull request:
- Экспортировал иконки
- Сверстал кнопку
- Добавил слушатель
- добавил круговой риппл эффект при нажатии на кнопку

KinCadeFoster committed:
RA-16
- Экспортировал иконки
- Сверстал кнопку
- Добавил слушатель
- добавил круговой риппл эффект при нажатии на кнопку

aGroupoid commented on app/src/main/java/ru/example/recipeapp/RecipeFragment.kt:
```diff
@@ -20,6 +21,9 @@ class RecipeFragment : Fragment() {
         FragmentRecipeBinding.inflate(layoutInflater)
     }
 
+    private lateinit var imageHeartButton: ImageButton
```
Глобальной эту переменную делать не надо, лучше её убрать вообще. Плюс при неправельном использовании может крашнут приложение , lateinit  лучше избегать, делать первичное значение хотя бы null

aGroupoid commented on app/src/main/java/ru/example/recipeapp/RecipeFragment.kt:
```diff
@@ -69,8 +73,19 @@ class RecipeFragment : Fragment() {
         }
+
+        imageHeartButton = binding.imageHeartButton
+        imageHeartButton.setOnClickListener {
```
сильно ли это делает код более читаемым?
Или инициализация глобальной переменной, присвоение, а затем только прикрепление listener? Почему бы сразу напрямую не обращаться к элементу через binding?

binding.imageHeartButton.setOnClickListener{}

KinCadeFoster replied:
нет, видимо фрагментное пошаговое мышление)

aGroupoid commented on app/src/main/res/layout/fragment_recipe.xml:
```diff
@@ -21,14 +21,28 @@
+            android:layout_height="wrap_content"
+            android:layout_marginTop="@dimen/common_12dp"
+            android:layout_marginEnd="@dimen/common_12dp"
+            android:background="@drawable/ripple_heart_button"
```
ripple эффект щас не работает, добавь те же параметры как для карточек. Их можно выести в отдельный стиль и переиспользовать

KinCadeFoster replied:
потыкал, но что то не то выходит) возможно это бага ОС

KinCadeFoster replied:
переделал, но почему то не работает, видимо чего то не понимаю

KinCadeFoster replied:
хотел поменять форму кнопки
    android:shape="oval"
но почему то всеравно квадратной остается =(

aGroupoid replied:
ты в foreground прописывал @drawable/ripple_heart_button? Вроде как должно работать так. Ладно, это не критично, главное, что на всех устройствах он есть)

aGroupoid reviewed (changes requested):
ic_heart_empty_button.png лишний файл, его нужно удалить

KinCadeFoster committed:
RA-16
- удалил ic_heart_empty_button.png
- удалил private lateinit var imageHeartButton: ImageButton
- рефакторинг кода по замечаниям

aGroupoid commented:
Перепроверил прошлую реализацию на своем девайсе, действительно работала. Но на дефолтных эмуляторах pixel она ничего не делает.
Предлагаю сделать так:
 android:foreground="?attr/selectableItemBackgroundBorderless" // можешь прописать старый рипл, должно работать
 android:background="#00000000" // пустой цвет, вынести в ресурсы
С такими параметрами работает и на пикселе, и на моем устройстве

KinCadeFoster committed:
RA-16
- поправил риппл эффект
- поправил баг, обрезания ползунка на seekbar

aGroupoid reviewed (approved):
Всё верно, молодец!
